OpenSim/Common/TimeSeriesTable.h

```diff
@@ -468,6 +477,31 @@ class TimeSeriesTable_ : public DataTable_<double, ETY> {
     void trimTo(const double& newFinalTime) {
         this->trim(this->getIndependentColumn().front(), newFinalTime);
     }
+    /**
+     * trim table to rows between start_index and last_index inclusively
+     * \throws InvalidIndices if the last index is out of range.
+     * \throws EmptyTable if the trim would make the table empty.
+     */
+    void trimToIndices(const size_t& start_index, const size_t& last_index) {
+        // Error if last_index is out of bounds
+        OPENSIM_THROW_IF(last_index > this->getNumRows() - 1, InvalidIndices);
```

Comment:
Should we also check if `start_index` is greater than -1? Or will `SimTK::Matrix` throw an exception for us here?

Reply:
I think because it is a size_t type it will not be negative, but I went ahead added a test case for it.

Reply:
Ah, yes. Good to test anyway.